packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart

```diff
@@ -49,26 +61,42 @@ abstract class InAppPurchaseConnection {
 
   /// Buy a non consumable product or subscription.
   ///
-  /// Non consumable items are the items that user can only buy once, for example, a purchase that unlocks a special content in your APP.
-  /// Subscriptions are also non consumable products.
-  ///
-  /// You need to restore all the non consumable products for user when they switch their phones.
-  ///
-  /// On iOS, you can define your product as a non consumable items in the [App Store Connect](https://appstoreconnect.apple.com/login).
-  /// Unfortunately, [Google Play Console](https://play.google.com/) defaults all the products as non consumable. You have to consume the consumable items manually calling [consumePurchase].
-  ///
-  /// This method does not return anything. Instead, after triggering this method, purchase updates will be sent to [purchaseUpdatedStream].
-  /// You should [Stream.listen] to [purchaseUpdatedStream] to get [PurchaseDetails] objects in different [PurchaseDetails.status] and
-  /// update your UI accordingly. When the [PurchaseDetails.status] is [PurchaseStatus.purchased] or [PurchaseStatus.error], you should deliver the content or handle the error, then call
-  /// [completePurchase] to finish the purchasing process.
-  ///
-  /// You can find more details on testing payments on iOS [here](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Chapters/ShowUI.html#//apple_ref/doc/uid/TP40008267-CH3-SW11).
-  /// You can find more details on testing payments on Android [here](https://developer.android.com/google/play/billing/billing_testing).
+  /// Non consumable items can only be bought once. For example, a purchase that
+  /// unlocks a special content in your app. Subscriptions are also non
+  /// consumable products.
+  ///
+  /// You always need to restore all the non consumable products for user when
+  /// they switch their phones.
+  ///
+  /// This method does not return anything. Instead, after triggering this
+  /// method, purchase updates will be sent to [purchaseUpdatedStream]. You
+  /// should [Stream.listen] to [purchaseUpdatedStream] to get [PurchaseDetails]
+  /// objects in different [PurchaseDetails.status] and update your UI
+  /// accordingly. When the [PurchaseDetails.status] is
+  /// [PurchaseStatus.purchased] or [PurchaseStatus.error], you should deliver
+  /// the content or handle the error, then call [completePurchase] to finish
+  /// the purchasing process.
+  ///
+  /// Consumable items are defined differently by the different underlying
+  /// payment platforms, and there's no way to query for whether or not the
+  /// [ProductDetail] is a consumable or not at runtime. On iOS, products are
+  /// defined as non consumable items in the [App Store
+  /// Connect](https://appstoreconnect.apple.com/login). [Google Play
+  /// Console](https://play.google.com/) products are considered consumable if
+  /// and when they are actively consumed manually.
```

Comment:
maybe:
consumed via [consume] manually

Reply:
It wouldn't be quite true, since they're consumed by the BillingClient consume method. I'm not sure if getting into that much detail here would be helpful. thinking about this more I think it would be helpful to make the lifecycle of a transaction clear in its own text somewhere. I'll try and do this as a followup.